### PR TITLE
Unify pseudo-random seeding interface

### PR DESCRIPTION
**Pull request by stefanv**
Use `rng` consistently, replacing `random_state` and `seed`.

See also https://github.com/scientific-python/specs/pull/180

Note:

- `BRIEF` features: seed is always set to 1. This means that the algorithm is
  deterministic, but what then is the point of having an RNG, and of
  exposing it to the user?
- For `medial_axis`, we use a random pixel assignment to break ties.
  It is not clear to me whether the randomness there is necessary, and
  especially whether it should be exposed via the API.
- Quickshift: seed is fixed to 42. Is this necessary?

Also /cc @jni who may be familiar with the seeding decision in Quickshift.

### CODE REVIEW

File: skimage/feature/brief.py [hunk 1]
```diff
@@ -130,10 +134,10 @@ def __init__(self, descriptor_size=256, patch_size=49,
         self.mode = mode
         self.sigma = sigma
 
-        if seed is None:
-            self.seed = np.random.SeedSequence().entropy
+        if rng is None:
+            self.seed = np.random.SeedSequence()
         else:
-            self.seed = seed
+            self.seed = rng
```
Comment (jarrodmillman) [inline skimage/feature/brief.py:140]: Is this needed? Shouldn't it just be
```
self.seed = rng
```
Also, should it be `self.rng`?

Comment (stefanv) [reply skimage/feature/brief.py:140]: No, because the rng changes as it gets used. The idea here is to re-seed every time `.extract` is called.

Comment (stefanv) [reply skimage/feature/brief.py:140]: I think this is correct. You want to re-use the same seed every time you run `.extract`. You can only do that if the seed is set to something other than None. So, if the RNG is None, then you grab a random seed once, and use it on every subsequent call of `.extract`.

**Review by jarrodmillman — APPROVED** [review]

**Comment by stefanv** [comment]
In general, I think this tie-breaker RNG approach is flawed, and could be removed without any adverse impact on the user.

**Comment by jarrodmillman** [comment]
We should deal with the tie-breaker issues in a separate PR. I am merging this since it is just about unifying the interface.

**Pull request merged**